pkg/server/commands/list_objects.go

```diff
@@ -288,6 +292,14 @@ func (q *ListObjectsQuery) evaluate(
 	return nil
 }
 
+func sendObject(object string, foundObjectsMap *sync.Map, foundCount *uint32, maxResults uint32, resultsChan chan<- ListObjectsResult) {
+	if _, loaded := foundObjectsMap.LoadOrStore(object, struct{}{}); !loaded {
+		if foundCount != nil && atomic.AddUint32(foundCount, 1) <= maxResults {
```

Comment:
Want to confirm that it is our intention where if maxResults == 0, we will not be sending the result to the resultsChan.  This is because if maxResults==0, foundCount will be nil.

Reply:
We will still send through the channel in that case. What will be different is the `ResultStatus` will be `NoFurtherEval`

Reply:
If `maxresults == 0` we send all the objects. I'll send a followup PR to add tests for this

Reply:
But if `maxresults==0`, `found==nil`.  The condition will be false and we will not execute line 298.

Reply:
If possible, let's add the test before completing this PR?

Reply:
> The condition will be false and we will not execute line 298.

Correct, that is the line where we return early because we've hit the limit :)

Reply:
but there should be no limit when maxresults==0 though